src/Pos/Txp/Core/Tx.hs

```diff
@@ -53,21 +53,25 @@ topsortTxs toTx input =
             let tx = toTx a
             ifM (HS.member (whHash tx) <$> use tsVisited)
                 (tsUnprocessed %= tail)
-                (dfs2 HS.empty a tx)
+                (dfs2 HS.empty a)
             dfs1
     -- Does dfs putting vertices into tsResult in reversed order of
     -- visiting. visitedThis is map of visited vertices for _this_ dfs
-    -- (cycle detection).
-    dfs2 :: HashSet (Hash Tx) -> a -> WithHash Tx -> State (TopsortState a) ()
-    dfs2 visitedThis _ (WithHash _ txHash)
-        | txHash `HS.member` visitedThis = tsLoop .= True
-    dfs2 visitedThis a (WithHash tx txHash) = unlessM (use tsLoop) $ do
-        tsVisited %= HS.insert txHash
-        let visitedNew = HS.insert txHash visitedThis
-            dependsUnfiltered =
-                mapMaybe (\x -> HM.lookup (txInHash x) txHashes) (tx ^. txInputs . to toList)
-        depends <- filterM
-            (\x -> not . HS.member (whHash (toTx x)) <$> use tsVisited)
-            dependsUnfiltered
-        forM_ depends $ \a' -> dfs2 visitedNew a' (toTx a')
-        tsResult %= (a:)
+    -- (used for loop detection).
+    dfs2 :: HashSet (Hash Tx) -> a -> State (TopsortState a) ()
```

Comment:
So it was an optimization? I considered it to be a very bad design because there was undocumented dependency between two function arguments. 
Also i think that this kind of optimizations are irrelevant for now.